qutebrowser/browser/webkit/network/pac.py

```diff
@@ -0,0 +1,608 @@
+    def dnsDomainLevels(self, host):
+        """Count number of DNS domain levels.
+
+        Return the number (integer) of DNS domain levels (number of dots)
+        in the hostname.
+
+        Args:
+            host: the hostname from the URL.
+        """
+        return host.count('.')
+
+    @_js_slot(str, str)
+    def shExpMatch(self, mstr, shexp):
+        """Test is string matches shell expression.
+
+        Return true if the string matches the specified shell expression.
+        Currently, the patterns are shell expressions, not regular
+        expressions.
+
+        Args:
+            mstr: Any string to compare (e.g. the URL, or the hostname).
+            shexp: a shell expression to compare against.
+        """
+        return fnmatch.fnmatchcase(mstr, shexp)
+
+    def _get_dt(self, gmt):
+        if self._fixed_dt is not None:
+            if gmt:
+                return self._fixed_dt.toUTC()
+            else:
+                return self._fixed_dt
+        else:
+            if gmt:
+                return QDateTime.currentDateTimeUtc()
+            else:
+                return QDateTime.currentDateTime()
+
+    @staticmethod
+    def _in_range(avals, vals, bvals, ranges):
+        """Test if a tuple is between two tuples in numerical order."""
+        carry_a = 0
+        carry_val = 0
+        carry_b = 0
+        for a, val, b, r in zip(avals, vals, bvals, ranges):
+            if a is not None:
+                a_c = a + carry_a
+                val_c = val + carry_val
+                b_c = b + carry_b
+                if a_c < val_c < b_c:
+                    return True
+                elif not a_c <= val_c <= b_c:
+                    return False
+                carry_a += a * r
+                carry_val += val * r
+                carry_b += b * r
+        return True
+
+    @_js_slot(QVariant)
+    def weekdayRange(self, args):
+        """Test is current date is between two week days.
+
+        Only the first parameter is mandatory. Either the second, the third,
+        or both may be left out.
+
+        If only one parameter is present, the function returns a value of
+        true on the weekday that the parameter represents. If the string
+        "GMT" is specified as a second parameter, times are taken to be in
+        GMT. Otherwise, they are assumed to be in the local timezone.
+
+        If both wd1 and wd1 are defined, the condition is true if the
+        current weekday is in between those two ordered weekdays. Bounds are
+        inclusive, but the bounds are ordered. If the "GMT" parameter is
+        specified, times are taken to be in GMT. Otherwise, the local
+        timezone is used.
+        """
+        weekdays = ("MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN")
+
+        # Expects list of all arguments because the function has different
+        # modes expecting different number of them.
+        if isinstance(args, QJSValue):
+            args = args.toVariant()
```

Comment:
I had this happen with new version of PyQt (on the local machine). Actually IIRC I initially had this conversion unconditionally and was very confused when Travis failed.